docs/book/en/src/dev/rust/wasinn.md

```diff
@@ -0,0 +1,158 @@
+
+```shell
+rustup target add wasm32-wasi
+```
+
+## Run
+
+Download the demo with:
+
+```sh
+git clone https://github.com/gusye1234/Wasi-NN-Examples.git
+```
+
+The demo will download the [fixtures](https://github.com/intel/openvino-rs/tree/main/crates/openvino/tests/fixtures/mobilenet) for the [Mobilenet](https://arxiv.org/abs/1704.04861), and run a image classification using the WASI-NN api:
+
+```sh
+cd Wasi-NN-Examples
+./build.sh <WASMEDGE RUNTIME PATH>
+# for example: 
+# ./build.sh "<path to WasmEdge repo>/build/tools/wasmedge/wasmedge"
+```
+
+If everything goes well, you should have the terminal output:
+
+```sh
+# massive rust compiling output above...
+Read graph XML, size in bytes: 143525
+Read graph weights, size in bytes: 13956476
+Loaded graph into wasi-nn with ID: 0
+Created wasi-nn execution context with ID: 0
+Read input tensor, size in bytes: 602112
+Executed graph inference
+   1.) [963](0.7113)pizza, pizza pie
+   2.) [762](0.0707)restaurant, eating house, eating place, eatery
+   3.) [909](0.0364)wok
+   4.) [926](0.0155)hot pot, hotpot
+   5.) [567](0.0153)frying pan, frypan, skillet
+```
+
+The generated `.wasm` file is under `./rust/examples/classification-example/build/wasi-nn-example.wasm`, we can see it imports the necessary WASI-NN functions using `wasm2wat`:
+
+```sh
+$ wast2wat ./rust/examples/classification-example/build/wasi-nn-example.wasm | grep import
+	(import "wasi_ephemeral_nn" "load" (func $_ZN7wasi_nn9generated17wasi_ephemeral_nn4load17hdca997591f45db43E (type 8)))
+  (import "wasi_ephemeral_nn" "init_execution_context" (func $_ZN7wasi_nn9generated17wasi_ephemeral_nn22init_execution_context17h2cb3b4398c18d1fdE (type 4)))
+  (import "wasi_ephemeral_nn" "set_input" (func $_ZN7wasi_nn9generated17wasi_ephemeral_nn9set_input17h4d10422433f5c246E (type 7)))
+  (import "wasi_ephemeral_nn" "get_output" (func $_ZN7wasi_nn9generated17wasi_ephemeral_nn10get_output17h117ce8ea097ddbebE (type 8)))
+  (import "wasi_ephemeral_nn" "compute" (func $_ZN7wasi_nn9generated17wasi_ephemeral_nn7compute17h96ef5b407fe8173aE (type 5)))
+  ...
+```
+
+## Code walkthrough
+
+Before running the actual ML inference, we need to download the necesary pre-processed artifacts for the demo:
+
+* `mobilenet.xml`: the model description
+* `mobilenet.bin`: the model weights
+* `tensor-1x3x224x224-f32.bgr`: the input image tensor.
+
+The above artifacts are generated by [OpenVINO™ Model Optimizer](https://docs.openvino.ai/2021.4/openvino_docs_MO_DG_Deep_Learning_Model_Optimizer_DevGuide.html). Thanks for the amazing jobs done by Andrew Brown, you can find the artifacts and a `build.sh` which can regenerate the artifacts [here](https://github.com/intel/openvino-rs/tree/main/crates/openvino/tests/fixtures/mobilenet). The demo will automatically download the ML artifacts and place them under `./rust/examples/classification-example/build/`(`tensor-1x3x227x227-f32.bgr` is renamed to `tensor.bgr`).
```

Comment:
Not to rename the `bgr` file name. The file name is for the information of the image format.